How to use the rif relay sample d app sdk (#946)
rsksmart/devportal: content/rsk-devportal/guides/rif-relay/deployment.md

      │          RelayVerifier                │ '0x03F23ae1917722d5A27a2Ea0Bcc98725a2a2a49a' │
      │        CustomSmartWallet              │ '0x1eD614cd3443EFd9c70F04b6d777aed947A4b0c4' │
      │    CustomSmartWalletFactory           │ '0x5159345aaB821172e795d56274D0f5FDFdC6aBD9' │
      │ CustomSmartWalletDeployVerifier       │ '0x7557fcE0BbFAe81a9508FF469D481f2c72a8B5f3' │
      │ CustomSmartWalletRelayVerifier        │ '0x0e19674ebc2c2B6Df3e7a1417c49b50235c61924' │
      │      NativeHolderSmartWallet          │ '0x4aC9422c7720eF71Cb219B006aB363Ab54BB4183' │
      │    NativeHolderSmartWalletFactory     │ '0xBaDb31cAf5B95edd785446B76219b60fB1f07233' │
      │ NativeHolderSmartWalletDeployVerifier │ '0xAe59e767768c6c25d64619Ee1c498Fd7D83e3c24' │
      │ NativeHolderSmartWalletRelayVerifier  │ '0x5897E84216220663F306676458Afc7bf2A6A3C52' │
      │            UtilToken                  │ '0x1Af2844A588759D0DE58abD568ADD96BB8B3B6D8' │
      │         VersionRegistry               │ '0x8901a2Bbf639bFD21A97004BA4D7aE2BD00B8DA8' │
      └───────────────────────────────────────┴──────────────────────────────────────────────┘
      ```

>     The deployment summary shows two sets of Smart Wallets, each paired with its verifiers. This is because each verifier uses the factorY for deployment and relay validation. For testing purposes, the focus will be on using these Smart Wallet Contracts.

Reviewer: ```suggestion
    The deployment summary shows two sets of Smart Wallets, each paired with its verifiers. This is because the verifier is used for both deployment and transaction validation. For testing purposes, the focus will be on using these Smart Wallet Contracts.
```

                  "beneficiary": "0xCF7CDBbB5F7BA79d3ffe74A0bBA13FC0295F6036",
                  "share": 13
                },
                {
                  "beneficiary": "0x39B12C05E8503356E3a7DF0B7B33efA4c054C409",
                  "share": 32
                }
            ],
            "tokenAddresses": ["0x1Af2844A588759D0DE58abD568ADD96BB8B3B6D8"],
            "remainderAddress": "0xc354D97642FAa06781b76Ffb6786f72cd7746C97"
          }
          ```
          > **Note:** The `collectorOwner`, `beneficiaries`, and `remainderAddress` are the first five accounts provided by the node in Regtest.
    2. Deploy the contract:
        ```bash
        npx hardhat collector:deploy --network regtest
        ```
        The collector is ready and can start receiving fees.
- Testnet
    - Using the configuration file you created in the regtest section, run this command to deploy the contract: 
        ```
        npx hardhat collector:deploy --network testnet
        ```
- Mainnet
    - Using the configuration file you created in the regtest section, run this command to deploy the contract: 
        ```
        npx hardhat collector:deploy --network mainnet
        ```

> ### Allow Tokens

Reviewer: Could you please review the allow-tokens command to reflect the latest changes included in the [repo](https://github.com/rsksmart/rif-relay-contracts?tab=readme-ov-file#managing-tokens)?

RIF Relay only accepts whitelisted tokens, primarily to ensure only tokens of value to the sponsor are accepted. To whitelist a token:
Execute the `acceptToken(address token)` function on the Relay Verifiers contracts, which include:
- `SmartWalletDeployVerifier`
- `SmartWalletRelayVerifier`

> Note: This action must be performed by the contracts' owner, typically the account that conducted the deployment.
  
[](#top "collapsible")
- Regtest
    - In the RIF Relay Contracts, execute this command:
        ```

>         npx hardhat allow-tokens --network regtest 0x1Af2844A588759D0DE58abD568ADD96BB8B3B6D8

Reviewer: This command should have the additional params `--token-list` and `--verifier-list`

        ```
        > The `allowTokens` uses the first account (referred to as account[0]) as the owner of the contracts. This is important because only the account owner can allow tokens. 
- Testnet
    - In the RIF Relay Contracts, execute the command:
        ```

>         npx hardhat allow-tokens --network testnet 0x3F49BaB0afdC36E9f5784da91b32E3D5156fAa5C

Reviewer: This command should have the additional params `--token-list` and `--verifier-list`

        ```
        > The `allowTokens` script will use the Testnet network configured in the `hardhat.config.ts`, this network will be required to use the account that deployed the contracts.
- Mainnet
    - In the RIF Relay Contracts, execute the command:
      ```

>       npx hardhat allow-tokens --network mainnet 0xe49b8906A3ceFd184621A4193e2451b1c3C3dB0B

Reviewer: This command should have the additional params `--token-list` and `--verifier-list`

                  "share": 13
                },
                {
                  "beneficiary": "0x39B12C05E8503356E3a7DF0B7B33efA4c054C409",
                  "share": 32
                }
            ],
            "tokenAddresses": ["0x1Af2844A588759D0DE58abD568ADD96BB8B3B6D8"],
            "remainderAddress": "0xc354D97642FAa06781b76Ffb6786f72cd7746C97"
          }
          ```
          > **Note:** The `collectorOwner`, `beneficiaries`, and `remainderAddress` are the first five accounts provided by the node in Regtest.
    2. Deploy the contract:
        ```bash
        npx hardhat collector:deploy --network regtest
        ```
        The collector is ready and can start receiving fees.
- Testnet
    - Using the configuration file you created in the regtest section, run this command to deploy the contract: 
        ```
        npx hardhat collector:deploy --network testnet
        ```
- Mainnet
    - Using the configuration file you created in the regtest section, run this command to deploy the contract: 
        ```
        npx hardhat collector:deploy --network mainnet
        ```
### Allow Tokens

> RIF Relay only accepts whitelisted tokens, primarily to ensure only tokens of value to the sponsor are accepted. To whitelist a token:

Reviewer: ```suggestion
RIF Relay only accepts whitelisted tokens, primarily to ensure only tokens of value to the operator are accepted. To whitelist a token:
```

Execute the `acceptToken(address token)` function on the Relay Verifiers contracts, which include:
- `SmartWalletDeployVerifier`
- `SmartWalletRelayVerifier`

> **Note:** This action must be performed by the contracts' owner, typically the account that conducted the deployment.
  
[](#top "collapsible")
- **Regtest**
    - In the RIF Relay Contracts, execute this command:
        ```
        npx hardhat allow-tokens --network regtest --token-list <TOKEN_ADDRESSES>
        ```
        > `<TOKEN_ADDRESSES>` is a comma-separated list of the token addresses to be allowed on the available verifiers. The `allowTokens` uses the first account (referred to as account[0]) as the owner of the contracts. This is important because only the account owner can allow tokens. 
- **Testnet**
    - In the RIF Relay Contracts, execute the command:
        ```

>         npx hardhat allow-tokens --network testnet --token-list <TOKEN_ADDRESSES>

Reviewer: Here we don't say that there is also the possibility to change the allowed tokens for certain verifiers only, see [managing tokens](https://github.com/rsksmart/rif-relay-contracts?tab=readme-ov-file#managing-tokens) in the contract repo. That's why I don't like repeating the same docs, because we'll need to maintain both.

        ```
        > `<TOKEN_ADDRESSES>` is a comma-separated list of the token addresses to be allowed on the available verifiers. The `allowTokens` script will use the Testnet network configured in the `hardhat.config.ts`, this network will be required to use the account that deployed the contracts.
- **Mainnet**
    - In the RIF Relay Contracts, execute the command:
      ```

>       npx hardhat allow-tokens --network mainnet --token-list <TOKEN_ADDRESSES>

Reviewer: Same as above. Actually in mainnet it's also more important since we want to save some money by changing only the used verifiers

        ```
        > **Note:** Relay and Deploy verifiers use the contracts from the Smart Wallet section in the summary.
        The meaning of each key can be found in [RIF Relay Server Configuration](https://github.com/rsksmart/rif-relay-server#server-configuration)
    1. To start the server, run the following command:
        ```
        npm run start
        ```
        > By default, the server uses the `default.json5` file in the config directory. Depending on the profile in `NODE_ENV`, the values in the `default.json5` file is overriden.
        At this point the server should be running and ready to start processing transactions, however, you still need to register the off-chain components in the Relay Hub.
    1. To enable the server for transaction processing, you must register the off-chain components in the Relay Hub. This step requires the server to be active. To register the components, in a different terminal window, execute the following command:
        ```
        npm run register
        ```
        The register process performs the following actions:
        - Stakes the Relay Manager
        - Adds the Relay Worker
        - Registers the Relay Server

>         The server is now ready to start processing transactions and a ready message is diplayed on the console. For more details on configuration and registration parameters, refer to the [RIF Relay Server documentation](https://github.com/rsksmart/rif-relay-server#overrides).

Reviewer: ```suggestion
        The server is now ready to start processing transactions and a `ready` message is diplayed on the console. For more details on configuration and registration parameters, refer to the [RIF Relay Server documentation](https://github.com/rsksmart/rif-relay-server#overrides).
```

            },
            "contracts": {
                "relayHubAddress": "0xAd525463961399793f8716b0D85133ff7503a7C2",
                "relayVerifierAddress": "0xB86c972Ff212838C4c396199B27a0DBe45560df8",
                "deployVerifierAddress": "0xc67f193Bb1D64F13FD49E2da6586a2F417e56b16"
            }
        }
        ```
        > The [contracts](https://developers.rsk.co/rif/relay/contracts/#primary-contracts) used in this setup are the primary contracts available on the Rootstock network. These primary contracts, however, do not include support for the `CustomSmartWallet`.            
        For details of each configuration key used in setting up the RIF Relay Server, refer to the [RIF Relay Server Configuration](https://github.com/rsksmart/rif-relay-server#server-configuration) documentation.
    1. To start the server, execute the following command:
        ```
        npm run start
        ```
        > By default, the server uses the `default.json5` file in the config directory. Depending on the profile in `NODE_ENV`, the values in the `default.json5` file is overriden. Therefore you need to setup the `NODE_ENV` environment to `testnet`.

>         At this point the server should be running and ready to start processing transactions, however, you still need to register the off-chain components in the Relay Hub. For the registration process, the Relay Manager and Worker must have funds.           

Reviewer: ```suggestion
        At this point, the server should be running and ready to start processing transactions; however, you still need to register the off-chain components in the Relay Hub. For the registration process, the Relay Manager and Worker must have funds.           
```

>     1. To get the addresses, this requires the server to be active. In a different terminal window, execute the following command:

Reviewer: ```suggestion
    1. To get the addresses, it requires the server to be active. In a different terminal window, execute the following command:
```

          "version": "2.0.1"
        }
        ```

>     1. Send at least 0.001 tRBTC to the Worker and Manager.

Reviewer: The `0.001` amount is arbitrary and configurable.
